beacon_node/beacon_chain/tests/electra.rs

```diff
@@ -0,0 +1,106 @@
+#![cfg(not(debug_assertions))] // Tests run too slow in debug.
+
+use beacon_chain::test_utils::BeaconChainHarness;
+use state_processing::{
+    per_block_processing, BlockSignatureStrategy, ConsensusContext, VerifyBlockRoot,
+};
+use std::sync::Arc;
+use types::*;
+
+type E = MainnetEthSpec;
+
+#[tokio::test]
+async fn signature_verify_chain_segment_pubkey_cache() {
+    let initial_validator_count = 32;
+
+    let deposit_slot = Slot::new(4 * E::slots_per_epoch() - 1);
+    let pre_deposit_slot = deposit_slot - 1;
+    let spec = Arc::new(ForkName::Electra.make_genesis_spec(E::default_spec()));
+
+    let harness = BeaconChainHarness::builder(E::default())
+        .spec(spec.clone())
+        .logger(logging::test_logger())
+        .deterministic_keypairs(initial_validator_count)
+        .fresh_ephemeral_store()
+        .mock_execution_layer()
+        .build();
+
+    harness
+        .execution_block_generator()
+        .move_to_terminal_block()
+        .unwrap();
+
+    Box::pin(harness.extend_to_slot(pre_deposit_slot)).await;
+
+    // Create a block with a deposit for a new validator.
+    let pre_deposit_state = harness.get_current_state();
+    assert_eq!(pre_deposit_state.slot(), pre_deposit_slot);
+    assert_eq!(pre_deposit_state.fork_name_unchecked(), ForkName::Electra);
+
+    // FIXME: Probably need to make this deterministic?
+    let new_keypair = Keypair::random();
+    let withdrawal_credentials = Hash256::ZERO;
+    let amount = spec.max_effective_balance_electra;
+    let deposit_data = harness.make_deposit_data(&new_keypair, withdrawal_credentials, amount);
+    let deposit_request = DepositRequest {
+        pubkey: deposit_data.pubkey,
+        withdrawal_credentials: deposit_data.withdrawal_credentials,
+        amount: deposit_data.amount,
+        signature: deposit_data.signature,
+        index: 0,
+    };
+
+    let ((jank_block, blobs), mut state) = harness
+        .make_block_with_modifier(pre_deposit_state, deposit_slot, |block| {
+            block
+                .body_mut()
+                .execution_requests_mut()
+                .unwrap()
+                .deposits
+                .push(deposit_request)
+                .unwrap();
+        })
+        .await;
+
+    // Compute correct state root.
+    // FIXME: this is kinda nasty
```

Comment:
it probably would have been cleaner to use the execution block generator to include the deposit requests, but it's a PITA in my experience, so I opted for just hacking the deposit into the block and recomputing the state root and signature